src/routing/approved.rs

```diff
@@ -450,19 +426,19 @@ impl Approved {
                     return Ok(MessageStatus::Useless);
                 }
             }
-            Variant::DKGResult { .. } => {
-                if !self.is_elder() {
-                    return Ok(MessageStatus::Unknown);
-                }
-            }
             Variant::NodeApproval(_) | Variant::BootstrapResponse(_) => {
                 // Skip validation of these. We will validate them inside the bootstrap task.
                 return Ok(MessageStatus::Useful);
             }
-            Variant::Vote { proof_share, .. } => {
-                if !self.should_handle_vote(proof_share) {
-                    // Message will be bounced if we are lagging (not known of the signing key).
-                    return Ok(MessageStatus::Unknown);
+            Variant::Vote {
+                content,
+                proof_share,
+                ..
+            } => {
+                if let Some(status) =
+                    self.decide_vote_status(&msg.src().to_node_name()?, content, proof_share)
```

Comment:
this is bit strange that a returned `Some` is actually a `useless/unknown` message.

Reply:
The idea is that when this returns `Some` then that's the status to be used, but when it returns `None` that means the status wasn't decided and we need to fall back to the default status processing. What would you suggest to make this more clear?

Reply:
I'd prefer to return explicit `MessageStatus::...` below, to be consistent with other branches.
though, not a big preference, up to you.